Add Windows CI

It seems currently on Windows, there are 11 failing tests.

There should be test on Windows.

```
> linkinator@0.0.0 pretest C:\Users\xmr\Desktop\linkinator
> npm run compile


> linkinator@0.0.0 compile C:\Users\xmr\Desktop\linkinator
> tsc -p .


> linkinator@0.0.0 test C:\Users\xmr\Desktop\linkinator
> nyc mocha build/test



  linkinator
    1) should perform a basic shallow scan
    2) should only try a link once
    3) should skip links if asked nicely
    4) should report broken links
    5) should handle relative links
    √ should handle fetch exceptions
    6) should skip mailto: links
    √ should report malformed links as broken
    7) should detect broken image links
    8) should perform a recursive scan
    9) should not recurse non-html files
    10) should not recurse by default
    11) should retry with a GET after a HEAD


  2 passing (126ms)
  11 failing

  1) linkinator
       should perform a basic shallow scan:

      AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert.ok(results.passed)

      + expected - actual

      -false
      +true

      at Context.it (test\test.ts:21:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  2) linkinator
       should only try a link once:

      AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert.ok(results.passed)

      + expected - actual

      -false
      +true

      at Context.it (test\test.ts:30:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  3) linkinator
       should skip links if asked nicely:

      AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert.ok(results.passed)

      + expected - actual

      -false
      +true

      at Context.it (test\test.ts:40:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  4) linkinator
       should report broken links:

      AssertionError [ERR_ASSERTION]: Mocks not yet satisfied:
HEAD http://fake.local:80/
      + expected - actual

      -false
      +true

      at Scope.done (node_modules\nock\lib\scope.js:155:10)
      at Context.it (test\test.ts:57:11)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  5) linkinator
       should handle relative links:

      AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert.ok(results.passed)

      + expected - actual

      -false
      +true

      at Context.it (test\test.ts:65:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  6) linkinator
       should skip mailto: links:

      AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert.ok(results.passed)

      + expected - actual

      -false
      +true

      at Context.it (test\test.ts:83:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  7) linkinator
       should detect broken image links:

      AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- 1
+ 2
      + expected - actual

      -1
      +2

      at Context.it (test\test.ts:101:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  8) linkinator
       should perform a recursive scan:

      AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- 1
+ 5
      + expected - actual

      -1
      +5

      at Context.it (test\test.ts:121:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  9) linkinator
       should not recurse non-html files:

      AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- 1
+ 2
      + expected - actual

      -1
      +2

      at Context.it (test\test.ts:130:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  10) linkinator
       should not recurse by default:

      AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- 1
+ 2
      + expected - actual

      -1
      +2

      at Context.it (test\test.ts:135:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  11) linkinator
       should retry with a GET after a HEAD:

      AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert.ok(results.passed)

      + expected - actual

      -false
      +true

      at Context.it (test\test.ts:148:12)
      at process._tickCallback (internal/process/next_tick.js:68:7)



----------|----------|----------|----------|----------|-------------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------|----------|----------|----------|----------|-------------------|
All files |        0 |        0 |        0 |        0 |                   |
----------|----------|----------|----------|----------|-------------------|
npm ERR! Test failed.  See above for more details.

```

--- a/.ci/docker/windows/Dockerfile
+++ b/.ci/docker/windows/Dockerfile
@@ -1,0 +1,2 @@
+FROM cirrusci/windowsservercore:2019
+RUN choco install nodejs --yes --version 10.15.3

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -1,6 +1,13 @@
 container:
   image: node:10
 
+windows_task:
+  windows_container:
+    dockerfile: .ci/docker/windows/Dockerfile
+    os_version: 2019
+  install_script: npm install
+  test_script: npm test
+
 lint_task:
   install_script: npm install
   test_script: npm run lint

--- a/package.json
+++ b/package.json
@@ -21,19 +21,22 @@
   "dependencies": {
     "chalk": "^2.4.2",
     "cheerio": "^1.0.0-rc.2",
-    "ecstatic": "^4.1.2",
+    "finalhandler": "^1.1.2",
     "gaxios": "^2.0.0",
     "jsonexport": "^2.4.1",
     "meow": "^5.0.0",
+    "serve-static": "^1.14.1",
     "server-destroy": "^1.0.1",
     "update-notifier": "^3.0.0"
   },
   "devDependencies": {
     "@types/cheerio": "^0.22.10",
+    "@types/finalhandler": "^1.1.0",
     "@types/meow": "^5.0.0",
     "@types/mocha": "^5.2.6",
     "@types/nock": "^10.0.0",
     "@types/node": "^11.9.4",
+    "@types/serve-static": "^1.13.2",
     "@types/server-destroy": "^1.0.0",
     "@types/sinon": "^7.0.6",
     "@types/update-notifier": "^2.5.0",
@@ -42,7 +45,7 @@
     "mocha": "^6.1.4",
     "nock": "^10.0.6",
     "nyc": "^14.0.0",
-    "semantic-release": "^15.13.12",
+    "semantic-release": "^15.13.13",
     "sinon": "^7.2.3",
     "source-map-support": "^0.5.10",
     "typescript": "^3.3.3"

--- a/src/index.ts
+++ b/src/index.ts
@@ -5,7 +5,8 @@ import enableDestroy = require('server-destroy');
 
 import { getLinks } from './links';
 
-const ecstatic = require('ecstatic');
+const finalhandler = require('finalhandler');
+const serveStatic = require('serve-static');
 
 export interface CheckOptions {
   port?: number;
@@ -83,10 +84,12 @@ export class LinkChecker extends EventEmitter {
    * @returns Promise that resolves with the instance of the HTTP server
    */
   private startWebServer(root: string, port: number): Promise<http.Server> {
-    return new Promise(resolve => {
+    return new Promise((resolve, reject) => {
+      const serve = serveStatic(root);
       const server = http
-        .createServer(ecstatic({ root }))
-        .listen(port, () => resolve(server));
+        .createServer((req, res) => serve(req, res, finalhandler(req, res)))
+        .listen(port, () => resolve(server))
+        .on('error', reject);
     });
   }
 

